crates/hyperswitch_connectors/src/utils.rs

```diff
@@ -2241,3 +2241,20 @@ impl WalletData for hyperswitch_domain_models::payment_method_data::WalletData {
         }
     }
 }
+
+pub fn deserialize_xml_to_struct<T: serde::de::DeserializeOwned>(
+    xml_data: &[u8],
+) -> Result<T, errors::ConnectorError> {
+    let response_str = std::str::from_utf8(xml_data)
+        .map_err(|e| {
+            router_env::logger::error!("Error converting response data to UTF-8: {:?}", e);
+            errors::ConnectorError::ResponseDeserializationFailed
+        })?
+        .trim();
+    let result: T = quick_xml::de::from_str(response_str).map_err(|e| {
+        router_env::logger::error!("Error deserializing XML response: {:?}", e);
```

Comment:
Is there a possibility of this logging sensitive information?

Reply:
I have checked the response; we don’t receive any connector credentials or card details in it.